app/controllers/application_controller.rb

```diff
@@ -437,10 +437,16 @@ def sp_session
   end
 
   def sp_session_request_url_with_updated_params
-    # Login.gov redirects to the orginal request_url after a user authenticates
-    # replace prompt=login with prompt=select_account to prevent sign_out
-    # which should only every occur once when the user lands on Login.gov with prompt=login
-    url = sp_session[:request_url]&.gsub('prompt=login', 'prompt=select_account')
+    return unless sp_session[:request_url].present?
+    request_url = URI(sp_session[:request_url])
+    url = if request_url.path.match?('saml')
+            complete_saml_url
+    else
```

Comment:
**nitpick:** echoing @zachmargolis' comment about the indentation (this and the following lines until `end` should line up with the `if`)

Reply:
https://github.com/18F/identity-idp/pull/6894/commits/c60ce446df4f3fbe1e5c9b557a886373d1864407 there we go